analyses/cell-type-wilms-tumor-14/scripts/utils/00_preprocessing_rds_functions.R

```diff
@@ -0,0 +1,84 @@
+library(dplyr)
+library(Seurat)
+library(ggpubr)
+pre_seuratobj <- function(obj, nfeatures = 500, run_harmony = TRUE, reduction = "harmony", ndims = 50){
+  ######## Normalize, scale, feature selection
+  obj <- Seurat::NormalizeData(obj, normalization.method = "LogNormalize")
+  obj <- Seurat::FindVariableFeatures(obj, selection.method = "vst", nfeatures = nfeatures)
+  obj <- Seurat::ScaleData(obj, features = Seurat::VariableFeatures(object = obj))
+  # obj <- Seurat::SCTransform(obj)
+  obj <- Seurat::RunPCA(obj, features = Seurat::VariableFeatures(object = obj))
+  # Seurat::ElbowPlot(obj, ndims = 50)
+  
+  ######## batch effect correction
+  if (run_harmony){
+    obj <- harmony::RunHarmony(obj, group.by.vars = "library_id")
+  }
+  
+  ######## Clustering and dimentional reduction
+  obj <- Seurat::FindNeighbors(obj, dims = 1:ndims, reduction = reduction)
+  obj <- Seurat::FindClusters(obj, resolution = 0.8, algorithm = 1)
+  obj <- Seurat::RunUMAP(obj, dims = 1:ndims, reduction = reduction)
+  
+  return(obj)
+}
+
+## remove duplicated genes in the list for dot plot
+clean_gslist <- function(inlist, obj){
+  dupgenes <- unique(unlist(inlist)[duplicated(unlist(inlist))])
+  availgenes <- rownames(obj)
+  for (i in 1:length(inlist)) {
+    element <- inlist[[i]][inlist[[i]] %in% availgenes]
+    element <- element[!element %in% dupgenes]
+    inlist[[i]] <- element
+  }
+  return(inlist)
+}
+
+## create and process a seurat object from provided sce object
+process_sce <- function( sample, library,
+                         path_proj = "/home/lightsail-user/git/OpenScPCA-analysis/data/current/SCPCP000014",
+                         db_proj = "/home/lightsail-user/git/OpenScPCA-analysis/data/current/results/doublet-detection/SCPCP000014",
+                         path_anal = "/home/lightsail-user/git/OpenScPCA-analysis/analyses/cell-type-wilms-tumor-14/"
+){
+  
+  ########### per sample rds preprocessing ##########
+  
+  #sample <- "SCPCS000517"; library <- "SCPCL000849"
+  rds <- readRDS(paste0(path_proj,"/",sample,"/",library,"_processed.rds"))
+  # out_rds <- paste0(path_proj,"/",sample,"/",library,"_processed_genesym.rds")
+  db <- read.table(paste0(db_proj,"/",sample,"/",library,"_processed_scdblfinder.tsv"), header = T)
+  
+  ######## convert ensembl IDs to symbols and dedup
+  rownames(rds) <- rds@rowRanges@elementMetadata@listData[["gene_symbol"]]
+  rds <- singleCellTK::dedupRowNames(rds, as.rowData = F, return.list = F)
+  rds <- rds[!is.na(rownames(rds)),]
+  rds <- rds[!duplicated(rownames(rds)),]
+  # readr::write_rds(rds, out_rds, compress = "bz2")
+  
+  ######## remove doublets
+  rds <- rds[,which(db$class == "singlet")]
```

Comment:
I don't think I have a very strong preference about the final resolution at this point, as I view this as an empirical question. My main thought is just that preserving some flexibility so that we can look more carefully at the differences later may be beneficial.